CONTRIBUTING.md

```diff
@@ -301,6 +301,36 @@ with an environment variable:
 JUJU_NOTEST_MONGOJS=1 go test github.com/juju/juju/...
 ```
 
+Conventional commits
+--------------------
+
+Once you have written some code and have tested the changes, the next step is to
+`git commit` it. For commit messages Juju follows the conventional commits guidelines specified [in this
+document](https://docs.google.com/document/d/1SYUo9G7qZ_jdoVXpUVamS5VCgHmtZ0QA-wZxKoMS-C0/).
+In short the commits should be of the following form:
+```
+<type>(optional <scope>): <description>
+
+[optional body]
+
+[optional footer(s)]
+```
+- **Type:** The type describes the kind of change (e.g., feat, fix, docs, style,
+  refactor, test, chore).
+- **Scope:** The scope indicates the part of the codebase affected (e.g., model,
+  api, cli).
+- **Description:** The description briefly describe the change. It should not
+  end in any punctuation.
```

Comment:
I'm not sure if the whole thing needs to be lower-case, but the first letter does need to be at least.

```suggestion
- **Description:** The description briefly describe the change. It should not
  end in any punctuation, and must be in lower-case.
```

Do you think it's good or bad to define "briefly"? It's 100 characters (I'm not sure if that's the description only or the entire first line including type and scope), and you do get told that in the error message if you are not brief enough.

Reply:
Ah yea, good point, I've never thought about the line length because vim hightlights it for you when writing, but I think that should be in there.

I've just put in an extra paragraph to talk about the whole first line.

Reply:
Acc, just realised that vim is keeping me on 72 not 100.

Reply:
@SimonRichardson Can we not change our implementation of conventional commits to say 72 recommended? If its a pain to enforce we can still just have it as a guideline. Currently our public doc does not have any guidance on the line length.

Reply:
Yeah, I'm fine with saying a guideline of 72 is preferable.